### DISCUSSION TITLE
Compile Nested JSON File Parsing

### DISCUSSION THREAD

**Discussion by GODFATHER23** [nlohmann/json#2478] [Q&A] [answered]
### While parsing complex nested JSON file it is difficult to access/search value of the object.

**Here is the json file which I'm working on --**

{
  "statuses": [
    {
      "metadata": {
        "result_type": "recent",
        "iso_language_code": "ja"
      },
      "created_at": "Sun Aug 31 00:29:15 +0000 2014",
      "id": 505874924095815700,
      "id_str": "505874924095815681",
      "text": "@aym0566x \n\n名前:前田あゆみ\n第一印象:なんか怖っ！\n今の印象:とりあえずキモい。噛み合わない\n好きなところ:ぶすでキモいとこ😋✨✨\n思い出:んーーー、ありすぎ😊❤️\nLINE交換できる？:あぁ……ごめん✋\nトプ画をみて:照れますがな😘✨\n一言:お前は一生もんのダチ💖",
      "source": "<a href=\"http://twitter.com/download/iphone\" rel=\"nofollow\">Twitter for iPhone</a>",
      "truncated": false,
      "in_reply_to_status_id": null,
      "in_reply_to_status_id_str": null,
      "in_reply_to_user_id": 866260188,
      "in_reply_to_user_id_str": "866260188",
      "in_reply_to_screen_name": "aym0566x",
      "user": {
        "id": 1186275104,
        "id_str": "1186275104",
        "name": "AYUMI",
        "screen_name": "ayuu0123",
        "location": "",
        "description": "元野球部マネージャー❤︎…最高の夏をありがとう…❤︎",
        "url": null,
        "entities": {
          "description": {
            "urls": []
          }
        }
       }
     }
   ]
}

**Using the sample code of your's I was able to parse whole json file but while searching for an object and it's value I wasn't able to search it just shows no such object/object value.**
Please help me with the code as in how and what to do.

**Comment by nlohmann** [comment, marked as answer]
You can iterate the object just like a `std::map`.

**Reply by chrom11** [reply]
Hi nlohmnann,
Thanks for the insight. I was just about to post a question about accessing nested data types. Would you mind please adding a link to the documentation that covers how to access those nested values or a quick code snippet? I'm a newbie and unfamiliar with std::map.

[Here is a link to the JSON data I'm using.](https://api.coincap.io/v2/assets) The content I want to get is all under "data": [ ] and I want to return the size and contents of the stuff within.

I'd be happy to open a new discussion showing my code and what's not working if this doesn't apply to the original poster's question.
Thanks in advance!

**Reply by nlohmann** [reply]
See https://nlohmann.github.io/json/features/iterators/ for an overview.
